{"name": "Login with Invalid Password", "status": "failed", "statusDetails": {"message": "AssertionError: assert False", "trace": "self = <testcases.test_login.Test_001_Login object at 0x1028b74d0>\n\n    @allure.title(\"Login with Invalid Password\")\n    @allure.severity(allure.severity_level.CRITICAL)\n    def test_invalid_password(self):\n        self.logger.info(\"********** Testing Invalid Password Login **********\")\n        self.lp = LoginPage(self.driver)\n        try:\n>           self.lp.login(self.username, \"invalid_password\")\n\ntestcases/test_login.py:44: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npageobjects/LoginPage.py:43: in login\n    self.enter_username(username)\npageobjects/LoginPage.py:19: in enter_username\n    WebDriverWait(self.driver, 10).until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"9ba5765db0b44b988040324b83b9d309\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x102b1b380>, message = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Wait until the method returns a value that is not False.\n    \n        Calls the method provided with the driver as an argument until the\n        return value does not evaluate to ``False``.\n    \n        Parameters:\n        -----------\n        method: callable(WebDriver)\n            - A callable object that takes a WebDriver instance as an argument.\n    \n        message: str\n            - Optional message for :exc:`TimeoutException`\n    \n        Return:\n        -------\n        object: T\n            - The result of the last call to `method`\n    \n        Raises:\n        -------\n        TimeoutException\n            - If 'method' does not return a truthy value within the WebDriverWait\n            object's timeout\n    \n        Example:\n        --------\n        >>> from selenium.webdriver.common.by import By\n        >>> from selenium.webdriver.support.ui import WebDriverWait\n        >>> from selenium.webdriver.support import expected_conditions as EC\n    \n        # Wait until an element is visible on the page\n        >>> wait = WebDriverWait(driver, 10)\n        >>> element = wait.until(EC.visibility_of_element_located((By.ID, \"exampleId\")))\n        >>> print(element.text)\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       0   chromedriver                        0x000000010533ea54 cxxbridge1$str$ptr + 2803960\nE       1   chromedriver                        0x0000000105336cf0 cxxbridge1$str$ptr + 2771860\nE       2   chromedriver                        0x0000000104e82864 cxxbridge1$string$len + 93028\nE       3   chromedriver                        0x0000000104ec9410 cxxbridge1$string$len + 382736\nE       4   chromedriver                        0x0000000104f0a480 cxxbridge1$string$len + 649088\nE       5   chromedriver                        0x0000000104ebd7ec cxxbridge1$string$len + 334572\nE       6   chromedriver                        0x0000000105303ccc cxxbridge1$str$ptr + 2562928\nE       7   chromedriver                        0x0000000105306f98 cxxbridge1$str$ptr + 2575932\nE       8   chromedriver                        0x00000001052e42c4 cxxbridge1$str$ptr + 2433384\nE       9   chromedriver                        0x0000000105307810 cxxbridge1$str$ptr + 2578100\nE       10  chromedriver                        0x00000001052d52f0 cxxbridge1$str$ptr + 2371988\nE       11  chromedriver                        0x000000010532757c cxxbridge1$str$ptr + 2708512\nE       12  chromedriver                        0x0000000105327708 cxxbridge1$str$ptr + 2708908\nE       13  chromedriver                        0x000000010533693c cxxbridge1$str$ptr + 2770912\nE       14  libsystem_pthread.dylib             0x0000000188605c0c _pthread_start + 136\nE       15  libsystem_pthread.dylib             0x0000000188600b80 thread_start + 8\n\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/selenium/webdriver/support/wait.py:146: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <testcases.test_login.Test_001_Login object at 0x1028b74d0>\n\n    @allure.title(\"Login with Invalid Password\")\n    @allure.severity(allure.severity_level.CRITICAL)\n    def test_invalid_password(self):\n        self.logger.info(\"********** Testing Invalid Password Login **********\")\n        self.lp = LoginPage(self.driver)\n        try:\n            self.lp.login(self.username, \"invalid_password\")\n            error_message = WebDriverWait(self.driver, 10).until(\n                EC.presence_of_element_located(\n                    (By.XPATH, \"//*[@id='app']/div[1]/div/div[1]/div/div[2]/div[2]/div/div[1]/div[1]/p\"))\n            )\n            assert \"Invalid credentials\" in error_message.text\n            self.logger.info(\"Invalid password test PASSED\")\n        except Exception as e:\n            self._handle_test_failure(\"invalid_password\", e)\n>           assert False\nE           assert False\n\ntestcases/test_login.py:53: AssertionError"}, "attachments": [{"name": "invalid_password_screenshot", "source": "e8bc9920-763b-4128-9c88-f494692658b0-attachment.png", "type": "image/png"}, {"name": "log", "source": "e67a50d9-aa8f-44cd-a874-d69244646736-attachment.txt", "type": "text/plain"}], "start": 1746551940901, "stop": 1746551951268, "uuid": "a6af5172-5688-4972-afa1-45a5d38a7431", "historyId": "aba918a04925777e096b92283807eaf8", "testCaseId": "aba918a04925777e096b92283807eaf8", "fullName": "testcases.test_login.Test_001_Login#test_invalid_password", "labels": [{"name": "severity", "value": "critical"}, {"name": "tag", "value": "@pytest.mark.usefixtures('setup')"}, {"name": "parentSuite", "value": "testcases"}, {"name": "suite", "value": "test_login"}, {"name": "subSuite", "value": "Test_001_Login"}, {"name": "host", "value": "viveks-MacBook-Air-2.local"}, {"name": "thread", "value": "1727-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcases.test_login"}]}